packages/ui/src/machines/authenticator/__tests__/defaultService.test.ts

```diff
@@ -98,6 +98,23 @@ describe('validateFormPassword', () => {
     });
   });
 
+  it('validates each of special characters allowed by Cognito as expected', async () => {
+    // adapted from https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-policies.html
+    const specialChars =
+      '^ $ * . [ ] { } ( ) ? " ! @ # % & / \\ , > < \' : ; | _ ~ ` = + -';
+    const specialCharsArray = specialChars.split(' ');
+    specialCharsArray.push(' '); // space is a valid character too (see docs above)
+
+    for (const char of specialCharsArray) {
```

Comment:
Rather than use a `for` loop here, we can place the values of `specialCharsArray` in an array and use the Jest idiomatic `it.each`, example:
```
// `%s` calls `toString` on the current value of `character`
it.each(['^', '$', '*', '.']).('validates usage of a %s character as expected`, (character) => {
  // test `character`
});
```

Reply:
Done in https://github.com/aws-amplify/amplify-ui/pull/2075/commits/82aa3a51934939428496da41d9eea05870f22dc8